e2e_test/source_inline/kafka/avro/partial_schema.slt

```diff
@@ -0,0 +1,199 @@
+
+# create topic and sr subject
+system ok
+rpk topic create 'avro_partial_schema_test'
+
+# create a schema
+system ok
+sr_register avro_partial_schema_test-value AVRO <<< '{"type":"record","name":"Root","fields":[{"name":"bar","type":"int","default":0},{"name":"foo","type":"string"}]}'
+
+# Specify schema
+statement ok
+create table t1 (foo varchar, bar int)
+WITH (
+    ${RISEDEV_KAFKA_WITH_OPTIONS_COMMON},
+    topic = 'avro_partial_schema_test'
+)
+FORMAT PLAIN ENCODE AVRO (
+    schema.registry = '${RISEDEV_SCHEMA_REGISTRY_URL}'
+);
+
+# Specify partial schema
+statement ok
+create table t2 (bar int, gen_col int as bar + 1)
+WITH (
+    ${RISEDEV_KAFKA_WITH_OPTIONS_COMMON},
+    topic = 'avro_partial_schema_test'
+)
+FORMAT PLAIN ENCODE AVRO (
+    schema.registry = '${RISEDEV_SCHEMA_REGISTRY_URL}'
+);
+
+# Specify incorrect schema
+statement error
+create table t (bar int, foo varchar, baz int)
+WITH (
+    ${RISEDEV_KAFKA_WITH_OPTIONS_COMMON},
+    topic = 'avro_partial_schema_test'
+)
+FORMAT PLAIN ENCODE AVRO (
+    schema.registry = '${RISEDEV_SCHEMA_REGISTRY_URL}'
+);
+----
+db error: ERROR: Failed to run the query
+
+Caused by:
+  Protocol error: Column "baz" is defined in SQL but not found in the source
+
+
+statement error
+create table t (bar double)
+WITH (
+    ${RISEDEV_KAFKA_WITH_OPTIONS_COMMON},
+    topic = 'avro_partial_schema_test'
+)
+FORMAT PLAIN ENCODE AVRO (
+    schema.registry = '${RISEDEV_SCHEMA_REGISTRY_URL}'
+);
+----
+db error: ERROR: Failed to run the query
+
+Caused by:
+  Protocol error: Data type mismatch for column "bar". Defined in SQL as "double precision", but found in the source as "integer"
+
+
+# Resolve schema
+statement ok
+create table tstar (*, gen_col int as bar + 1)
+WITH (
+    ${RISEDEV_KAFKA_WITH_OPTIONS_COMMON},
+    topic = 'avro_partial_schema_test'
+)
+FORMAT PLAIN ENCODE AVRO (
+    schema.registry = '${RISEDEV_SCHEMA_REGISTRY_URL}'
+);
+
+# No wildcard will be interpreted as `*` for syntax backward compatibility
+statement ok
+create table tstar2 (gen_col int as bar + 1)
+WITH (
+    ${RISEDEV_KAFKA_WITH_OPTIONS_COMMON},
+    topic = 'avro_partial_schema_test'
+)
+FORMAT PLAIN ENCODE AVRO (
+    schema.registry = '${RISEDEV_SCHEMA_REGISTRY_URL}'
+);
+
+
+# Demonstrate purified definition
+query TT rowsort
+SELECT name, SUBSTRING(definition, 1, POSITION(' WITH' IN definition) - 1) FROM rw_tables WHERE name LIKE 't%';
+----
+t1	    CREATE TABLE t1 (foo CHARACTER VARYING, bar INT)
+t2	    CREATE TABLE t2 (bar INT, gen_col INT AS bar + 1)
+tstar	CREATE TABLE tstar (bar INT, foo CHARACTER VARYING, gen_col INT AS bar + 1)
+tstar2	CREATE TABLE tstar2 (bar INT, foo CHARACTER VARYING, gen_col INT AS bar + 1)
+
+# create a new schema
+system ok
+sr_register avro_partial_schema_test-value AVRO <<< '{"type":"record","name":"Root","fields":[{"name":"bar","type":"int","default":0},{"name":"foo","type":"string"}, {"name":"baz", "type":"double", "default":0}]}'
```

Comment:
Can you add one more test

1. declare schema (a int, b varchar)
2. create table with generated column (a int, b varchar, c int as a + 1)
3. create a newer version in schema registry (a int, b varchar, c int)
4. what does the purified SQL tend to be at the moment?

Reply:
Step 3 will not affect the purified SQL because we don't access (then test against) the external schema unless we are going to perform schema change.

However, this is still an interesting example, as I'm not sure about the behavior if we call `REFRESH` or `ADD COLUMN` afterwards. I'm afraid this has long been undefined behavior since we allow the definition of a generated column on a source. Will test it.

Reply:
Covered in 4cce72a149bccbea19aef2412d9b7e1cd2ffa33b